### PR TITLE
Data validation

### PR DESCRIPTION
**Pull request by fisyamaloy**
1) Some members of MessageType enum are renamed correctly.
2) Store user in DB with password hash.
3) New input field for email.
4) Data validation works during registration.

### CODE REVIEW

**Review by StasDash — APPROVED** [review]

File: Utility/Utility.Public/Include/Utility/UserDataValidation.hpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * @return true if email is valid.
+     * @details Email must: /
+     *   -have first char like letter; /
+     *   -maintain the following rule: /
+     *     Some letters + '@' + some letters + '.' + com/ru. /
+     * Example: email = 'Anton3_antonov4@epam.com'.
+     */
+    inline bool isEmailValid(const std::string& email)
+    {
+        const std::regex pattern(R"(([a-zA-Z]{1})(\w+@\w+\.)(?:com|ru))");
```
Comment (vDrWool) [inline Utility/Utility.Public/Include/Utility/UserDataValidation.hpp:33]: Do you sure to use std::regax? It is maybe too slow.

File: Utility/Utility.Public/Include/Utility/UserDataValidation.hpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     * @details Password must have: /
+     *   -At least one uppercase letter; /
+     *   -at least one lowercase letter; /
+     *   -at least one number; /
+     *   -minimum 8 characters. /
+     * Example: password = '1996a*nton-Anton%ov_34'.
+     */
+    inline bool isPasswordValid(const std::string& password)
+    {
+        const std::regex pattern("(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9]).{8,}");
```
Comment (vDrWool) [inline Utility/Utility.Public/Include/Utility/UserDataValidation.hpp:49]: Here too.

File: Utility/Utility.Public/Include/Utility/UserDataValidation.hpp (hunk 1 above)
Comment (fisyamaloy) [reply Utility/Utility.Public/Include/Utility/UserDataValidation.hpp:33]: > Do you sure to use std::regax? It is maybe too slow.

Yes, checking for validaty take place on the client-side and the server is not loaded with it. But if checking was on the server-side, regex wouldn't be so good way, I think

**Review by vDrWool — APPROVED** [review]

**Review by GuskoMaksim — APPROVED** [review]

**Comment by f1nal3** [comment]
@fisyamaloy looks good to me, feel free to merge.

**Comment by f1nal3** [comment]
Swap women from com|ru to 2-4 chars any.

**Pull request merged**